clang/lib/Lex/Preprocessor.cpp

```diff
@@ -905,6 +912,7 @@ void Preprocessor::Lex(Token &Result) {
     // This token is injected to represent the translation of '#include "a.h"'
     // into "import a.h;". Mimic the notional ';'.
     case tok::annot_module_include:
+    case tok::annot_repl_input_end:
```

Comment:
Can you explain this change?

Reply:
This change was used to solve a clang-repl issue.
```sh
Failed Tests (1):
  Clang :: Interpreter/cxx20-modules.cppm
```
It's possible a bug. IIUC, we should reset states like `ModuleDeclState` when we meet a `tok::annot_repl_input_end` token in preprocessor.

Reply:
CC @vgvassilev

Reply:
> IIUC, we should reset states like ModuleDeclState when we meet a tok::annot_repl_input_end token in preprocessor.

Quite possibly. This change was introduced by @ChuanqiXu9, maybe he has a better idea?

Reply:
I introduced  `Interpreter/cxx20-modules.cppm` but I don't have a feeling for change here. Just for its name, it looks slightly odd that we will still handle things after we see `annot_repl_input_end`. (It is an end, isn't it?)

Reply:
It reads a bit odd indeed. I am not sure what we can do here, too... Perhaps that's fine as it is for now...